pkg/resources/table_grant.go

```diff
@@ -159,20 +169,24 @@ func ReadTableGrant(data *schema.ResourceData, meta interface{}) error {
 	var builder snowflake.GrantBuilder
 	if onFuture {
 		builder = snowflake.FutureTableGrant(dbName, schemaName)
-		return readGenericGrant(data, meta, builder, true)
 	} else {
 		builder = snowflake.TableGrant(dbName, schemaName, tableName)
-		return readGenericGrant(data, meta, builder, false)
 	}
+
+	return readGenericGrant(data, meta, builder, onFuture)
 }
 
 // DeleteTableGrant implements schema.DeleteFunc
 func DeleteTableGrant(data *schema.ResourceData, meta interface{}) error {
-	dbName, schemaName, tableName, _, err := splitGrantID(data.Id())
+	grantID, err := grantIDFromString(data.Id())
+
```

Comment:
```suggestion
```
extra line makes this harder to read because it adds unnecessary space between the return args and where they are being used